src/port/portindex.tcl

```diff
@@ -107,6 +129,24 @@ proc _open_port {portinfo_name portdir absportdir port_options_name {subport {}}
     set portinfo(portdir) $portdir
 }
 
+proc write_deleted_ports {} {
+    global qindex list_of_parsed_ports
+    foreach port [array names qindex] {
+        if {[lsearch -exact $list_of_parsed_ports $port] == -1 && [info exists qindex($port)] && [string trim $port] != ""} {
```

Comment:
I doubt if `[info exists qindex($port)]` condition is necessary in `if` above. You are looping over qindex already.

Reply:
Yes, it is not needed. Thank you.